types/src/transaction/native_transaction.rs

```diff
@@ -0,0 +1,176 @@
+use alloc::vec::Vec;
+use core::fmt::{self, Display, Formatter};
+
+#[cfg(feature = "datasize")]
+use datasize::DataSize;
+#[cfg(any(feature = "testing", test))]
+use rand::Rng;
+#[cfg(feature = "json-schema")]
+use schemars::JsonSchema;
+use serde::{Deserialize, Serialize};
+
+use super::AuctionTransaction;
+#[cfg(doc)]
+use super::Transaction;
+#[cfg(any(feature = "testing", test))]
+use crate::testing::TestRng;
+use crate::{
+    bytesrepr::{self, FromBytes, ToBytes, U8_SERIALIZED_LENGTH},
+    RuntimeArgs,
+};
+
+const MINT_TRANSFER_TAG: u8 = 0;
+const AUCTION_TAG: u8 = 1;
+const RESERVATION_TAG: u8 = 2;
+
+/// A [`Transaction`] targeting native functionality.
+#[derive(Clone, Ord, PartialOrd, Eq, PartialEq, Hash, Serialize, Deserialize, Debug)]
+#[cfg_attr(feature = "datasize", derive(DataSize))]
+#[cfg_attr(
+    feature = "json-schema",
+    derive(JsonSchema),
+    schemars(description = "A Transaction targeting native functionality.")
+)]
+#[serde(deny_unknown_fields)]
+#[non_exhaustive]
+pub enum NativeTransaction {
+    /// Calls the `transfer` entry point of the mint to transfer `Motes` from a source purse to a
+    /// target purse.
+    ///
+    /// Requires the following runtime args:
+    ///   * "source": `URef`
+    ///   * "target": `URef`
+    ///   * "amount": `U512`
+    ///
+    /// The following optional runtime args can also be provided:
+    ///   * "to": `Option<AccountHash>`
+    ///   * "id": `Option<u64>`
+    #[cfg_attr(
+        feature = "json-schema",
+        schemars(
+            description = "Calls the `transfer` entry point of the mint to transfer `Motes` from
+            a source purse to a target purse."
+        )
+    )]
+    MintTransfer(RuntimeArgs),
+
+    /// A transaction targeting the auction.
+    Auction(AuctionTransaction),
+
+    /// A transaction reserving a future execution.
```

Comment:
I remember that in the past there were some hard issues to solve before implementing this i.e. how do price something in the future, how do we ensure we wouldn't queue more than we can reasonably execute in a future block, etc. Is there a plan for this?

Reply:
Again, I'm not sure of the details here yet.